editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -6120,12 +6128,46 @@ Node *CanvasItemEditorViewport::_make_texture_node_type(String texture_node_type
 	return node;
 }
 
+// Copied from Node3DEditorViewport. Only useful when `PREVIEW_3D_SCENE_BACKGROUND` is enabled.
+void CanvasItemEditorViewport::_project_settings_changed() {
+	// Update shadow atlas if changed.
+	const int shadowmap_size = ProjectSettings::get_singleton()->get("rendering/shadows/shadow_atlas/size");
+	const bool shadowmap_16_bits = ProjectSettings::get_singleton()->get("rendering/shadows/shadow_atlas/16_bits");
```

Comment:
I don't like how this code is copied. When this behavior is inevitably changed or added to, one will probably forget to also update `CanvasItemEditorViewport`. This behavior should instead be added to a method like `Viewport::load_from_project_settings` which will also be helpful to end-users. `viewport.cpp` already depends on `project_settings.h` so it's not a big deal dependency-wise.